### ISSUE TITLE
HTTP request failed with status: 404 when trying to connect to ooba

### ISSUE THREAD

**Issue by Bibrrr** [badgids/OpenKlyde#11]
Hello :)
I have an issue with this app when it comes to connection to ooba with openai api. 
I changed port to 5000 in text-default.json config file to match port with that from ooba. Bot turns on just fine, it also appears to connect to ooba. However when messaging bot in discord, I get this 
![obraz](https://github.com/badgids/OpenKlyde/assets/115658726/58dfd243-cbdd-4bad-a456-5d3ab095b735)

I'm trying to figure out if it has something to do with openklyde, or openai api extension itself. Extention loads without that /v1 at the end 
![obraz](https://github.com/badgids/OpenKlyde/assets/115658726/2a67f650-c594-4b42-b014-f89b5977d564)

**Comment by badgids** [comment]
Looks like your missing a trailing / after v1 for your API address.

**Comment by Bibrrr** [comment]
That's a silly typo... but it's still not working 
![obraz](https://github.com/badgids/OpenKlyde/assets/115658726/5bfa27af-cabb-41cc-b729-c3ff79653137)
I noticed that when you run koboldcpp there's an actual api website when you go to 127.0.0.1:5001/api. I assume there should be something similar on 127.0.0.1:5000/api, but there's nothing 
![issue](https://github.com/badgids/OpenKlyde/assets/115658726/231649ba-55d5-4ac3-900d-1a20d760199c)

**Comment by badgids** [comment]
It's been a minute since I used Ooba. Did you launch Textgen-UI with the  --openai extension?

**Comment by Bibrrr** [comment]
I completely forgot about /docs... sorry about that. 
I think I see where's the issue. Your bot is looking for localhost:5000/api/v1/model, while in ooba it's localhost:5000/v1/models

for text generation in kobold it's 127.0.0.1:5001/api/v1/generate while in ooba it's 127.0.0.1:5000/v1/chat/completions . That's why it's throwing 404 errors.

I will look into the code and check if changing those values helps

**Comment by Bibrrr** [comment]
It's getting somewhere !
I changed text-default config to the following 
![obraz](https://github.com/badgids/OpenKlyde/assets/115658726/21a553f2-6d2c-46ad-b3af-6ccc2f4bc01e)
and it succesfully sends a request to ooba and generates text. There's a new issue... It looks like it can't retrieve data from openai api. 
I tried multiple things. I mostly focued on line 380 in bot.py file. I changed **_data = llm_response['results'][0]['text']_** to **_data = llm_response['choices'][0]['object']_**  as it's suggests in ooba docs. Now I get this error
![bot py error](https://github.com/badgids/OpenKlyde/assets/115658726/61c288ee-5a69-40d7-a117-30061def5807)
I also tried to change like 380 to _**data = llm_response['choices'][0]['message']['content']**_ as per openai api docs. It errors out at message (also included in 2nd screenshot).

Further changes that I tried...
I tried to change **"generation"** in config to **"generation": "chat/completions"**. It gives HTML error 422. With no text generation from ooba

**Comment by badgids** [comment]
Yeah it looks like Ooba has changed some things in the API from when I first wrote this program last year.
